src/transformers/onnx/config.py

```diff
@@ -99,6 +99,7 @@ class OnnxConfig(ABC):
                 "end_logits": {0: "batch", 1: "sequence"},
             }
         ),
+        "semantic-segmentation": OrderedDict({"logits": {0: "batch", 1: "sequence"}}),
```

Comment:
@michaelbenayoun 

For `SemanticSegmenterOutput`, the output shape is like
```
logits (`torch.FloatTensor` of shape `(batch_size, config.num_labels, logits_height, logits_width)`)
```

but I don't know how to add the correct entry here 😢 . Could you help me 🙏 ?

Reply:
cc @NielsRogge as he might also know this image stuff.

Reply:
I am also confused with this in `ViTOnnxConfig`:

```python
    @property
    def inputs(self) -> Mapping[str, Mapping[int, str]]:
        return OrderedDict(
            [
                ("pixel_values", {0: "batch", 1: "sequence"}),
            ]
        )
```

how does the image input have this kind of input shape and axis names 😕

Reply:
@ydshieh the `sequence` axis name was indeed misleading and has been fixed in https://github.com/huggingface/transformers/pull/17976

Would you mind rebasing to incorporate that change?

Regarding your question with the axes, I think the following would make sense for this output:

```suggestion
        "semantic-segmentation": OrderedDict({"logits": {0: "batch", 1: "num_labels", 2: "height", 3: "width"}}),
```

Having said that, running the export with

```
python -m transformers.onnx --model=facebook/data2vec-vision-base --feature=semantic-segmentation scratch/onnx/
```

reveals that there is a problem with exporting the `adaptive_avg_pool2d` op:

```
RuntimeError: ONNX export failed: Couldn't export operator aten::adaptive_avg_pool2d
```

I believe this is the cause of your segmentation fault error and I found a few related PyTorch issues:

* https://github.com/onnx/tutorials/issues/63
* https://github.com/pytorch/pytorch/issues/42653

My reading of these issues is that this op is not currently supported in ONNX. Some people suggest implementing a [custom layer](https://github.com/pytorch/pytorch/issues/42653#issuecomment-1168816422), but I'm not really sure this will work with pretrained model weights (i.e. inference accuracy could be ruined).

My suggestion would be to remove `semantic-segmentation` as a supported feature for the `data2vec-vision` export, possibly with a comment about the lack of ONNX support for `adaptive_avg_pool2d`

WDYT?